streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -277,6 +277,7 @@ public void reset() {
         needsRestoring.clear();
         endOffsets.clear();
         needsInitializing.clear();
+        completedRestorers.clear();
```

Comment:
This is @linyli001 's fix.

Reply:
any quick hack fix for this bug? such as adding some tricky code to listeners, before next version release.